src/lib/flow/sol-flow-resolver-conffile.c

```diff
@@ -179,21 +206,18 @@ _resolver_conffile_get_module(const char *type)
         goto error;
     }
 
-    g_hash_table_insert(resolver_conffile_dlopens,
-        entry->name, entry);
-
 found:
     ret = resolve_module_type_by_component(type, entry->foreach);
-    SOL_NULL_CHECK_GOTO(ret, wipe_entry);
-
+    SOL_NULL_CHECK_GOTO(ret, error);
     return ret;
 
 error:
+    /* In case 'entry' was not the last one to be added. */
+    index = find_entry_by_match(&resolver_conffile_dlopens, entry);
+    assert(index >= 0);
```

Comment:
In the case that there's an associated entry but 'resolve_module_type_by_component()' fails.